config/cdc/postgres-entrypoint.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# This script replaces the default docker entrypoint for postgres in the
+# development environment.
+# Its job is to ensure postgres is properly configured to support the
+# Change Data Capture pipeline (by setting access permissions and installing
+# the replication plugin we use for CDC). Unfortunately the default
+# Postgres image does not allow this level of configurability so we need
+# to do it this way in order not to have to publish and maintain our own
+# Postgres image.
+#
+# This then, at the end, transfers control to the default entrypoint.
+
+set -e
+
+cdc_setup_hba_conf() {
+    # Ensure pg-hba is properly configured to allow connections
+    # to the replication slots.
+
+    PG_HBA="$PGDATA/pg_hba.conf"
+    if [ ! -f "$PG_HBA" ]; then
+        echo "DB not initialized. Postgres will take care of pg_hba"
+    fi
+
+    if [ "$(grep -c -E "^host\s+replication" "$PGDATA"/pg_hba.conf)" -ge 0 ]; then
```

Comment:
> Is it always part of the image?

It is supposed to always be there after the DB is initialized. But I think you are right this may fail if the DB is not initialized. Let me wipe my image and try that use case with the latest version.

Reply:
Fixed. Now it starts up properly if the DB is not initialized.

```
(.venv_build) ~ $ docker container logs -f  sentry_postgres                                                                                                    
Setting up Change Data Capture                                                                                                                                 
DB not initialized. Postgres will take care of pg_hba                                                                                                          
Connecting to api.github.com (192.30.255.117:443)                                                                                                              
writing to stdout                                                                                                                                              
-                    100% |********************************|  3646  0:00:00 ETA                                                                                
written to stdout                                                                                                                                              
wal2json installed                                                                                                                                             
********************************************************************************                                                                               
WARNING: POSTGRES_HOST_AUTH_METHOD has been set to "trust". This will allow                                                                                    
         anyone with access to the Postgres port to access your database without                                                                               
         a password, even if POSTGRES_PASSWORD is set. See PostgreSQL                                                                                          
         documentation about "trust":                                                                                                                          
         https://www.postgresql.org/docs/current/auth-trust.html                                                                                               
         In Docker's default configuration, this is effectively any other                                                                                      
         container on the same system.                                                                                                                         
                                                                                                                                                               
         It is not recommended to use POSTGRES_HOST_AUTH_METHOD=trust. Replace                                                                                 
         it with "-e POSTGRES_PASSWORD=password" instead to set a password in                                                                                  
         "docker run".                                                                                                                                         
********************************************************************************                                                                               
The files belonging to this database system will be owned by user "postgres".                                                                                  
This user must also own the server process.                                                                                                                    
                                                                                                                                                               
The database cluster will be initialized with locale "en_US.utf8".                   
```